app/products/calls/actions/calls.ts

```diff
@@ -263,3 +275,97 @@ export const setSpeakerphoneOn = (speakerphoneOn: boolean) => {
     InCallManager.setSpeakerphoneOn(speakerphoneOn);
     setSpeakerPhone(speakerphoneOn);
 };
+
+export const endCallAlert = async (serverUrl: string, intl: IntlShape, channelId: string) => {
+    const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+    if (!database) {
+        return;
+    }
+
+    const currentUser = await getCurrentUser(database);
+    if (!currentUser) {
+        return;
+    }
+
+    const channel = await getChannelById(database, channelId);
+    if (!channel) {
+        return;
+    }
+
+    const call = getCallsState(serverUrl).calls[channelId];
+    if (!call) {
+        return;
+    }
+
+    const numParticipants = Object.keys(call.participants).length || 0;
+    const isAdmin = isSystemAdmin(currentUser.roles);
+    const {formatMessage} = intl;
+
+    if (!isAdmin && currentUser.id !== call.creatorId) {
+        Alert.alert(
+            formatMessage({
+                id: 'mobile.calls_end_permission_title',
+                defaultMessage: 'Error',
+            }),
+            formatMessage({
+                id: 'mobile.calls_end_permission_msg',
+                defaultMessage: 'You do not have permission to end the call. Please ask the call creator to end the call.',
+            }));
+        return;
+    }
+
+    const endCall = async () => {
+        const client = NetworkManager.getClient(serverUrl);
+
+        let data: ApiResp;
+        try {
+            data = await client.endCall(channelId);
+        } catch (error) {
+            await forceLogoutIfNecessary(serverUrl, error as ClientError);
+            throw error;
+        }
+
+        return data;
+    };
+
+    const title = formatMessage({id: 'mobile.calls_end_call_title', defaultMessage: 'End call'});
+    let msg = formatMessage({
+        id: 'mobile.calls_end_msg_channel',
+        defaultMessage: 'Are you sure you want to end a call with {numParticipants} participants in {displayName}?',
+    }, {numParticipants, displayName: channel.displayName});
+
+    if (channel.type === General.DM_CHANNEL) {
+        const otherID = getUserIdFromDM(channel.name, currentUser.id);
+        const otherUser = await getUserById(database, otherID);
+        const {config, license} = await getCommonSystemValues(database);
+        const preferences = await queryPreferencesByCategoryAndName(database, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.NAME_NAME_FORMAT).fetch();
+        const displaySetting = getTeammateNameDisplaySetting(preferences, config, license);
```

Comment:
Wow, looks like we need lots of stuff just to display a message. Wondering if at some point we could consider a wrapping util function or something to ease this.

Reply:
Yah, this is a pain. I'd love to wrap it. Sometimes it happens in the `enhance` HOC and it's a bit easier, but this is definitely not ideal. Would love to know if there's an easier way I'm missing.